artifactory/utils/search.go

```diff
@@ -188,3 +190,15 @@ func SearchFiles(servicesManager artifactory.ArtifactoryServicesManager, spec *s
 	}
 	return
 }
+
+func ConvertArtifactsSearchDetailsToBuildInfoArtifacts(artifactsDetailsReader *content.ContentReader) ([]buildinfo.Artifact, error) {
+	var buildArtifacts []buildinfo.Artifact
+	for artifactSearchDetails := new(utils.ResultItem); artifactsDetailsReader.NextRecord(artifactSearchDetails) == nil; artifactSearchDetails = new(utils.ResultItem) {
+		artifact := artifactSearchDetails.ToArtifact()
+		if i := strings.LastIndex(artifact.Name, "."); i != -1 {
+			artifact.Type = artifact.Name[i+1:]
+		}
+		buildArtifacts = append(buildArtifacts, artifact)
+	}
+	return buildArtifacts, artifactsDetailsReader.GetError()
```

Comment:
Does `artifactsDetailsReader.GetError()` always return error since if no record is present it returns `io.EOF`